astropy/coordinates/sites.py

```diff
@@ -0,0 +1,203 @@
+            errmsg = ("Site not in database. Use ``get_site_names()`` "
+                      "to see available sites. Did you mean: '{}'?".format(
+                      "', '".join(close_names)))
+        else:
+            errmsg = 'Site not in database.'
+        raise KeyError(errmsg)
+
+    return _site_db[site_name.lower()]
+
+def get_site_names(full_list=True):
+    """
+    Get list of names of observatories for use with
+    `~astropy.coordinates.get_site`.
+
+    Parameters
+    ----------
+    full_list : bool
+        Show full list observatory names and aliases (True), or just the list
+        of names (False)? Default to True.
+
+    Returns
+    -------
+    names : list of str
+        List of valid observatory names
+
+    See Also
+    --------
+    get_site : gets the `~astropy.coordinates.EarthLocation` for one of the
+               sites this returns.
+    get_site_names : the list of sites names that this function can access
+    remove_site : remove a site by name from the list of sites
+
+    """
+    if _site_db is None:
+        _load_sites()
+
+    if full_list:
+        return sorted(_site_db.keys())
+    else:
+        return sorted(_site_names)
+
+def add_site(site_names, location):
+    """
+    Add a site to the list of available observatories.
+
+    Parameters
+    ----------
+    site_names : string or list of strings
+        Name of the observatory.  If a list, all names will be aliases for the
+        same ``location``, and the first will be taken as the canonical name.
+    location : `~astropy.coordinates.EarthLocation`
+        Location of the observatory
+
+    See Also
+    --------
+    get_site : gets the `~astropy.coordinates.EarthLocation` for one of the
+               sites this returns.
+    get_site_names : the list of site names that this function will add to
+    remove_site : remove a site by name from the list of sites
+
+    """
+    if _site_db is None:
+        _load_sites()
+
+    if not isinstance(location, EarthLocation):
+        raise ValueError('Location must be an EarthLocation.')
+
+    if isinstance(site_names, six.string_types):
+        site_names = [site_names]
+
+    firstnamedone = False
+    for name in site_names:
+        if name.lower() not in _site_db.keys():
+            _site_db[name.lower()] = location
+            if not firstnamedone:
+                _site_names.append(name)
+                firstnamedone = True
+        else:
+            raise KeyError('The site "{}" already exists at (longitude,latitude,'
+                           'elevation)={}'.format(name,
+                                         _site_db[name.lower()].to_geodetic()))
+
+def remove_site(site_name, remove_aliases=False):
+    """
+    Removes a site from the list of available observatories.
+
+    Parameters
+    ----------
+    site_name : string
+        Name of the observatory to remove
+    remove_aliases : bool
+        Also remove any aliases for the corresponding location
```

Comment:
I think if you're going to remove a site it should just always remove all aliases.  A separate function(s) could be added to remove(/add) aliases.